vaadin-date-picker-flow/src/main/resources/META-INF/resources/frontend/datepickerConnector.js

```diff
@@ -1,140 +1,146 @@
+                /* create regex
+                * regex will be the date, so that:
+                * - day-part is '(\d{1,2})' (1 or 2 digits),
+                * - month-part is '(\d{1,2})' (1 or 2 digits),
+                * - year-part is '(\d{4})' (4 digits)
+                *
+                * and everything else is left as is.
+                * For example, us date "10/20/2010" => "(\d{1,2})/(\d{1,2})/(\d{4})".
+                *
+                * The sorting part solves that which part is which (for example,
+                * here the first part is month, second day and third year)
+                *  */
+                datepicker.$connector.regex = testString.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&').replace(datepicker.$connector.dayPart.initial, "(\\d{1,2})").replace(datepicker.$connector.monthPart.initial, "(\\d{1,2})").replace(datepicker.$connector.yearPart.initial, "(\\d{4})");
+
+                datepicker.i18n.formatDate = function (date) {
+                    let rawDate = new Date(Date.UTC(date.year, date.month, date.day));
+                    return cleanString(rawDate.toLocaleDateString(locale, { timeZone: 'UTC' }));
+                };
+
+                datepicker.i18n.parseDate = function (dateString) {
+                    dateString = cleanString(dateString);
+
+                    if (dateString.length == 0) {
+                        return;
                     }
-                    return {
-                        day: datepicker.$connector.dayPart.value,
-                        month: datepicker.$connector.monthPart.value - 1,
-                        year: datepicker.$connector.yearPart.value
-                    };
-                }  else {
-                    return false;
-                }
-            };
 
-            if (inputValue === "") {
-                oldLocale = locale;
-            } else if (currentDate) {
+                    let match = dateString.match(datepicker.$connector.regex);
+                    if (match && match.length == 4) {
+                        for (let i = 1; i < 4; i++) {
+                            datepicker.$connector.parts[i-1].value = parseInt(match[i]);
+                        }
+                        return {
+                            day: datepicker.$connector.dayPart.value,
+                            month: datepicker.$connector.monthPart.value - 1,
+                            year: datepicker.$connector.yearPart.value
+                        };
+                    }  else {
+                        return false;
+                    }
+                };
+
+                if (inputValue === "") {
+                    oldLocale = locale;
+                } else if (currentDate) {
                     /* set current date to invoke use of new locale */
                     datepicker._selectedDate = new Date(currentDate.year, currentDate.month, currentDate.day);
-            }
-        }
+                }
+            })
```

Comment:
This line should end in `;`. This is the end of `datepicker.$connector.setLocale =` assignment.

Reply:
Fixed.